.github/workflows/playwright.yml

```diff
@@ -1,45 +1,61 @@
 name: Playwright Tests
 
-on: [push, pull_request]
+on: workflow_call
 
 jobs:
-  test:
+  acceptance:
     strategy:
+      fail-fast: false
       matrix:
+        os: [windows-latest, ubuntu-latest, macos-latest]
         node-version: [17.x, 18.x, 19.x]
-    timeout-minutes: 60
-    runs-on: macos-latest
+    runs-on: ${{ matrix.os }}
     steps:
-      - name: "Checkout"
-        uses: actions/checkout@v3
-
-      - name: Install pnpm
-        uses: pnpm/action-setup@v2.2.4
-
-      - name: Build with NodeJS ${{ matrix.node-version }}
-        uses: actions/setup-node@v3
+      - uses: actions/checkout@v3
+      - uses: pnpm/action-setup@v2.2.4
+      - uses: actions/setup-node@v3
         with:
           node-version: ${{ matrix.node-version }}
           cache: "pnpm"
+      - uses: actions/download-artifact@v3
+        with:
+          name: build-output
+          path: packages/pages/dist
 
-      - name: Install deps and build
-        run: pnpm i && pnpm run build
-
-      - name: Install again to link deps
-        run: pnpm i
+      # It's important to not run the build for @yext/pages on Windows.
+      # Currently the build does not work (on Windows) and will replace the
+      # downloaded build artifact without any warnings/errors.
+      - run: pnpm i --ignore-scripts
 
       - name: Install playwright browsers
         run: pnpm exec playwright install
-
+  
       - name: Run Playwright tests
         run: |
           pnpm --filter locations-site run playwright
           pnpm --filter multibrand-site run playwright
 
+      # This step will commit any new snapshots that weren't taken locally.
+      # i.e. snapshots for Windows and Linux machines
+      - name: Commit New Snapshots
+        uses: EndBug/add-and-commit@v9
+        if: ${{ failure() && github.ref_name != 'main' }}
+        with:
+          message: Commit new snapshots for ${{ matrix.os }}
+          add: playground/*
+          push: true
+          pull: --rebase --autostash
+          default_author: github_actions 
+
       - name: Upload Playwright Reports
         uses: actions/upload-artifact@v3
         if: always()
         with:
           name: playwright-reports
           path: playground/*/playwright-report/
-          retention-days: 30
+          retention-days: 30   
+
+      - name: Ensure build doesn't error
+        run: |
+          pnpm --filter locations-site run build-test-site
+          pnpm --filter multibrand-site run build-test-site
```

Comment:
We might want to add build-test-site scripts to the multibrand repo to build for the scoped brands as well (e.g. `pages build --scope sunglasses.rayban.com`)

Reply:
Also, you can merge these commands using `pnpm --filter "./playground/*" run build-test-site`. Or even match on regex if we end up having `build-test-site:oakley` or something (`pnpm --filter "./playground/*" run "/^build-test.*/"`).

The reason I didn't merge the commands for running the Playwright tests is because it tries executing them in parallel. So the first site is served and then the second set of tests sees that the port is in use and tries to use that for its tests, resulting in errors.

Reply:
Ah ok that's cool they run in parallel! I'll use the first I think the regex is overkill if we don't need it

Reply:
Ended up adding a matrix for the site so everything can be in parallel

Reply:
Sounds good. Can we still add the scoped build-test-site commands though?

Reply:
It's scoped to matrix.site right now, what do you have in mind?

Reply:
Right now, `build-test-site` in the multibrand site only does a top-level, unscoped build. We could also test for builds scoped to the individual brands (e.g. `pages build --scope sunglasses.rayban.com`)